internal/gatewayapi/route.go

```diff
@@ -314,12 +314,46 @@ func (t *Translator) processHTTPRouteRule(httpRoute *HTTPRouteContext, ruleIdx i
 		ruleRoutes = append(ruleRoutes, irRoute)
 	}
 
+	var sessionPersistence *ir.SessionPersistence
+	if rule.SessionPersistence != nil {
+		if rule.SessionPersistence.IdleTimeout != nil {
+			return nil, fmt.Errorf("idle timeout is not supported in envoy gateway")
+		}
+
+		switch {
+		case rule.SessionPersistence.Type == nil || // Cookie-based session persistence is default.
+			*rule.SessionPersistence.Type == gwapiv1.CookieBasedSessionPersistence:
+			sessionPersistence = &ir.SessionPersistence{
+				Cookie: &ir.CookieBasedSessionPersistence{
+					Name: *rule.SessionPersistence.SessionName,
```

Comment:
SessionName is optional https://github.com/kubernetes-sigs/gateway-api/blob/d330bc0e63bb607aa6ab48dfa4376b3e903b23e0/apis/v1/shared_types.go#L759
does Envoy create a unique name in case the value is empty or does EG need to generate one ?

Reply:
Oh, I missed that the SessionName is optional.
EG has to generate the one because the stateful session filter marks the name config as required.